### PR TITLE
Fix issue where paused annotation being set to false still leads to scaled objects/jobs being paused

### LINKED ISSUE #5215: Setting autoscaling.keda.sh/paused: "false" leads to ScaledObject/Job being paused
### Report

I noticed that setting 
```
autoscaling.keda.sh/paused: "false"
```
pauses the ScaledObject.

### Expected Behavior

I would expect setting `autoscaling.keda.sh/paused: "false"` to turn off the forced pausing behavior. 

### Actual Behavior

Instead, regardless of what `autoscaling.keda.sh/paused` is set to, the ScaledObject remains in the paused state. To resume the scaling behavior, I removed the annotation.

### Steps to Reproduce the Problem

1. add `autoscaling.keda.sh/paused: "false"` annotation to an existing, running ScaledObject/ScaledJob
2. run `kubectl get ScaledObject` and verify that the object is in the `paused: True` state

### Logs from KEDA operator

_No response_

### KEDA Version

2.12.0

### Kubernetes Version

1.27

### Platform

Google Cloud

### Scaler Details

Redis

### Anything else?

I just noticed based on the check here - https://github.com/kedacore/keda/blob/25dc205132d1de09a6156b2bc9ce623353b94fdb/controllers/keda/scaledobject_controller.go#L213 that this is expected behavior. Also, in the documentation this looks like expected behavior as the suggested action to resume the scaling is to remove the annotation - https://keda.sh/docs/2.12/concepts/scaling-deployments/#pause-autoscaling

That being said, the use of `true` in the documentation is confusing as it implies that switching to `false` should disable the pausing functionality.

I was going to submit a PR to check the value of the annotation but I'm curious what people think the desired behavior should be.

### PR DESCRIPTION
**Pull request by nappelson**
This PR adds "bool parsing" logic to the `paused` annotation. Keda docs suggests pausing scaledjobs/objects by setting the paused annotation to `true`. When I tried resuming the object by setting the annotation to `false`, I noticed the object remained paused.

Just for reference (https://go.dev/src/strconv/atob.go)
```lang=go
// ParseBool returns the boolean value represented by the string.
// It accepts 1, t, T, TRUE, true, True, 0, f, F, FALSE, false, False.
``` 

### Checklist

- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Tests have been added
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes #5215

### CODE REVIEW

**Comment by nappelson** [comment]
@zroubalik curious if you have thoughts on why the `TestPredictKubeGetMetrics` may be failing here... when I check out main directly and run the tests on my dev container i see that test fail as well. 

The `main-build` appears to be passing on CI so I'm not 100% sure what is going on. Let me know if I'm missing something obvious. Happy to talk to someone else too. Otherwise, I'll dig more into what could be happening here.

**Comment by zroubalik** [comment]
> @zroubalik curious if you have thoughts on why the `TestPredictKubeGetMetrics` may be failing here... when I check out main directly and run the tests on my dev container i see that test fail as well.
> 
> The `main-build` appears to be passing on CI so I'm not 100% sure what is going on. Let me know if I'm missing something obvious. Happy to talk to someone else too. Otherwise, I'll dig more into what could be happening here.

Most likely a flaky test, pls disregard this one.

**Comment by nappelson** [comment]
> /run-e2e internals **Update:** You can check the progress [here](https://github.com/kedacore/keda/actions/runs/7221267700)

@zroubalik thanks for running these - curious why they failed remotely. I also see one of the controller tests is failing. I'll fix that shortly!

**Comment by nappelson** [comment]
> > /run-e2e internals **Update:** You can check the progress [here](https://github.com/kedacore/keda/actions/runs/7221267700)
> 
> @zroubalik thanks for running these! I also see one of the controller tests is failing. I'll fix that and the e2e test shortly.

**Comment by zroubalik** [comment]
The problem seems to be in pause_scaledjob/pause_scaledjob_test.go let me rerun the test.

**Comment by nappelson** [comment]
> For unit tests it is as you said:
> 
> ```
> [FAIL] ScaledObjectController [It] scaledobject paused condition status changes to true on annotation
>   /__w/keda/keda/controllers/keda/scaledobject_controller_test.go:935
>   [FAIL] ScaledObjectController [It] deletes hpa when scaledobject has pause annotation
>   /__w/keda/keda/controllers/keda/scaledobject_controller_test.go:1132
> ```
> 
> Could you please also rebase your PR? We have merged some annotations based PR: #5282

@zroubalik this should be ready again. Appreciate your patience.

**Review by zroubalik — APPROVED** [review]
LGTM, thanks a lot for the contribution! Great job.

File: CHANGELOG.md [hunk 1]
```diff
@@ -51,7 +51,7 @@ To learn more about active deprecations, we recommend checking [GitHub Discussio
 
 ### New
 
-- **General**: TODO ([#XXX](https://github.com/kedacore/keda/issues/XXX))
+- **General**: Fix issue where paused annotation being set to false still leads to scaled objects/jobs being paused ([#5215](https://github.com/kedacore/keda/issues/5215))
```
Comment (zroubalik) [inline CHANGELOG.md:54]: Nit, could you please move this to Improvements section down below? (sorted alphabetically)

**Review by zroubalik — COMMENTED** [review]
Do you think you can also update docs for 2.13 mentioning that true/false works with the annotation?

**Comment by nappelson** [comment]
> Do you think you can also update docs for 2.13 mentioning that true/false works with the annotation?

https://github.com/kedacore/keda-docs/pull/1283 just created this

**Review by zroubalik — APPROVED** [review]

**Pull request merged**